### PR TITLE
Descriptions are now displayed on list of /membership /groupings /admin

### PR DESCRIPTION
**Pull request by JorWo**
# Ticket Link

[Groupings-533](https://uhawaii.atlassian.net/browse/GROUPINGS-533)

# List of squashed commits

- Added tooltips to truncated text in descriptions and group paths
- Added table column settings with local storage to save settings
- Added tablet and phone screen support
- Changed indentation of mainv2_3.css from 2 to 4 spaces to avoid Codacy error
- Replaced "flex-break" class in admin.html with "w-100"
- Created new custom directive tooltipOnTruncate and added new js file app.directives.js to hold directives
- Added Jasmine Tests, added new file app.directives.test.js
- Added popper.js to pom.xml

# Test Checklist

- [x] Unit Tests Passed:
- [x] Jasmine Tests Passed:
- [x] General Visual Inspection:

### CODE REVIEW

**Comment by JorWo** [comment]
I will add Jasmine Testing in another ticket.

**Review by mhodgesatuh — APPROVED** [review]
I don't see any issues.

**Comment by devgav** [comment]
Seems like there are some merge conflicts that need to be resolved in our main css file first

**Comment by devgav** [comment]
Also if you are going to do this split i would recommend you link the pr for the jasmine testing as a future reference. Ideally, it would be in the same pr request tho...

**Comment by fduckart** [comment]
Those Bootstrap width overrides... I thought that's how bootstrap did it already? Think I've done that without overrides. But maybe I'm misreading the applicability of this in the case where it's used, tho: https://getbootstrap.com/docs/4.0/utilities/sizing/

**Comment by fduckart** [comment]
In app.util.js, this strikes me as a tad drastic:
```
$(".dropdown").click(function(){
    $(".dropdown-toggle").dropdown();
});
```
But I dunno, maybe we want to go to that extreme.

**Comment by devgav** [comment]
> In app.util.js, this strikes me as a tad drastic:
> 
> ```
> $(".dropdown").click(function(){
>     $(".dropdown-toggle").dropdown();
> });
> ```
> 
> But I dunno, maybe we want to go to that extreme.

I def think that this dropdown functionality could be done with AngularJS instead of jquery. Would be easier to test as well.

**Comment by JorWo** [comment]
> > In app.util.js, this strikes me as a tad drastic:
> > ```
> > $(".dropdown").click(function(){
> >     $(".dropdown-toggle").dropdown();
> > });
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > But I dunno, maybe we want to go to that extreme.
> 
> I def think that this dropdown functionality could be done with AngularJS instead of jquery. Would be easier to test as well.

I just updated that code in app.util.js to better reflect the tooltip jquery function above it. I prefer to have it as a global function so that any button with data-toggle="dropdown" will toggle its dropdown menu.

**Comment by JorWo** [comment]
Just added jasmine testing and I put a comment to clarify this code:
> $scope.columnDisplaySetting = storedColumnSetting === null ? "first" : storedColumnSetting;
$scope.showDescriptionColumn = storedDescriptionColumn === null ? true : (storedDescriptionColumn === "true");
$scope.showPathColumn = storedPathColumn === null ? false : (storedPathColumn === "true");

**Comment by JorWo** [comment]
Just cleaned up this code:
>$scope.columnDisplaySetting = storedColumnSetting === null ? "first" : storedColumnSetting;
$scope.showDescriptionColumn = storedDescriptionColumn === null ? true : (storedDescriptionColumn === "true");
$scope.showPathColumn = storedPathColumn === null ? false : (storedPathColumn === "true");`

**Comment by devgav** [comment]
> > > In app.util.js, this strikes me as a tad drastic:
> > > ```
> > > $(".dropdown").click(function(){
> > >     $(".dropdown-toggle").dropdown();
> > > });
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > But I dunno, maybe we want to go to that extreme.
> > 
> > 
> > I def think that this dropdown functionality could be done with AngularJS instead of jquery. Would be easier to test as well.
> 
> I just updated that code in app.util.js to better reflect the tooltip jquery function above it. I prefer to have it as a global function so that any button with data-toggle="dropdown" will toggle its dropdown menu when clicked on.

As a team we are actively trying to reduce our use of jquery and use AngularJs instead. As far as your preference with using jquery as a global function AngularJs actually has their own solution by using $scope functions and adding a ng-model attribute to a html element so now any html element with the ng-model="scopefunctionname" can share that logic. I recommend you change both your dropdown and tooltip hover jquery solutions to AngularJs solutions.

**Comment by JorWo** [comment]
I have removed the jquery and fixed the dropdown menu by moving the popper.min.js import before bootstrap.min.js (Bootstrap's dropdowns use popper.js but it didn't seem to recognize the import). As for the tooltips, the ones I added in this PR does not use the jquery in app.util.js. I can make a new ticket removing the jquery from the other tooltips.

**Comment by JorWo** [comment]
I am closing the PR to fix subtle bugs, will reopen once fixed: 
- Missing tooltip arrow when using uib-tooltip
- Page scrolls to the top when pagination is used

**Review by devgav — APPROVED** [review]

File: src/main/resources/static/javascript/mainApp/app.util.js [hunk 1]
```diff
@@ -4,3 +4,7 @@ $(function () {
         container: "body"
     });
 });
+
+$(".dropdown").click(function(){
```
[the remark this answers is not in the document]
Comment (JorWo) [reply src/main/resources/static/javascript/mainApp/app.util.js]: I have already removed the jquery for the dropdown functionality. See my second to last previous comment.

**Pull request merged**